File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;
int b, y;

vector<int> solution(int brown, int yellow) {
    b = brown; y = yellow;
    vector<int> answer;
    
    int by = brown/2;
    int tmpC = 1;
    while(1) {
        int tmpR = by - tmpC;
        int r = tmpR+1; int c = tmpC+1;
        int total = r*c;
        if(total - brown == yellow) {
            answer.push_back(r); answer.push_back(c);
            break;
        }
        tmpC++;
        
    }
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        for(int h=2; h<=2000000; h++) {
            int w = (brown+4)/2 - h;
            if((h-2) * (w-2) == yellow) {
                answer[0] = w;
                answer[1] = h;
                break;
            }
        }
        
        return answer;
    }
}